### ISSUE TITLE
Attribute duplication via admin interface

### ISSUE THREAD

**Issue by BlauFeuer** [evennia/evennia#1009]
#### Brief summary of issue / Description of requested feature:

When using localhost:8000/admin/typeclasses/attribute/ to add or rename attributes,
the objectdb attribute is created with an empty Strvalue (instead of NULL), and editing
an attribute in-game via `@attribute` causes a duplicate attribute instead of updating
the existing one.
#### Steps to reproduce the issue / Reasons for adding feature:
1. Rename an existing attribute via admin interface.
2. `@set` the attribute in-game.
3. `examine` and observe duplicate.
#### Error output / Expected result of feature

`@set #4/desc_brief='A person on the move.'`

```
Created attribute Atrial/desc_brief = 'A person on the move.'
------------------------------------------------------------------------------
Name/key: Atrial (#4)
session(s): 29
Player: Wizard
Player Perms: <Superuser>
Typeclass: Character (typeclasses.characters.Character)
Location: Limbo (#2)
Permissions: players [Superuser]
Locks:call:false(); control:perm(Immortals); delete:perm(Wizards);
      edit:perm(Wizards); examine:perm(Builders); get:false(); hug:all();
      puppet:id(4) or pid(1) or perm(Immortals) or pperm(Immortals);
      tell:perm(Wizards); view:all()
Stored Cmdset(s):
commands.default_cmdsets.CharacterCmdSet [DefaultCharacter] (Union, prio 0)
Merged Cmdset(s):
evennia.commands.cmdset.CmdSet [_channelset] (Union, prio 120)
evennia.commands.cmdset.CmdSet [_exitset] (Union, prio 101)
evennia.commands.cmdset.CmdSet [_danceset] (Union, prio 0)
commands.default_cmdsets.PlayerCmdSet [DefaultPlayer] (Union, prio -10)
commands.default_cmdsets.SessionCmdSet [DefaultSession] (Union, prio -20)
Commands available to Atrial (result of Merged CmdSets):
@alias, @ban, @batchcode, @batchcommands, @boot, @charcreate, [...]
Persistent attributes:
desc = The invisible cloak of covering up.
prelogout_location = Limbo
combat_parry_mode = False
puzzle_clue = 0
coins = 12
desc_brief = 20000
traits = {'health': {'base': 20, 'name': 'Health', 'min': 0, 'max': 20, 'extra[...]
messages = {}
desc_brief = A person on the move.
Non-Persistent attributes:
last_cmd = None
last_location = Limbo
Tags: vegetarian
Contents: The Hawblade(#148)
```
#### Extra information, such as Evennia revision/repo/branch, operating system and ideas for how to solve / implement:

current version, main branch, any OS

```
```

**Comment by TehomCD** [comment]
After looking at the code for a while, I think it's a caching issue. AttributeHandler.add() checks the cache, and since django admin hasn't called _recache(), @set will still treat the attribute by its old key in the cache. AttributeHandler.all() returns all the attributes in the cache by their values, so its name is properly shown to be the new db_key. Related issues are needing more validation in django admin for Attribute change/creation to ensure there aren't duplicate names created. Should we assume that every Attribute will be in one, and only one, object? I know it's a ManyToMany relation, but from db_model field specifying the model its attached to, I get the impression they're more supposed to be used like a foreignkey to a typeclass.

**Comment by BlauFeuer** [comment]
I suspect that having to `@reload` after making edits in the admin interface specific to adding or removing command sets on objects is related to not recaching when needed, so if this attribute duplication problem is solved by updating cache, so, too, would many other caching related problems.

**Comment by TehomCD** [comment]
Maybe a post_save signal in Objects and Attributes that recache the cmdsethandler and associated object's attributehandler, respectively?

**Comment by Griatch** [comment]
This is resolved with the merger of #1115. Closing.
